thread/common/omrthread.c

```diff
@@ -3307,7 +3321,10 @@ omrthread_park(int64_t millis, intptr_t nanos)
 	ASSERT(self);
 
 #if defined(OMR_THR_YIELD_ALG)
-	omrthread_park_spin(self, millis, nanos, &sleptDuration);
+	J9ThreadLibrary *threadLibrary = self->library;
```

Comment:
This is a C function, so declarations need to be placed at the start of the function, before any code is executed. In this case, the declaration should go before `ASSERT(self)`. The best place would be inside the `#ifdef` below `sleptDuration`.

Use
```
omrthread_library_t lib = self->library;
```

instead of `J9ThreadLibrary`; we want to avoid introducing additional instances of `J9`-prefixed variable names in the OMR repo.